src/main/java/org/broadinstitute/hellbender/tools/funcotator/DataSourceFuncotationFactory.java

```diff
@@ -37,6 +39,16 @@
      */
     protected Map<String, String> annotationOverrideMap;
 
+    /**
+     * The backing data store as a FeatureInput to leverage tribble querying.  Can be {@code null} for non-locatable
+     * funcotation factories.
+     */
+    protected FeatureInput<? extends Feature> mainSourceFileAsFeatureInput;
```

Comment:
Can this be `final`, since it's initialized in the subclasses at construction time?

Reply:
Sure.  I created a couple of constructors in `DataSourceFuncotationFactory` handle the setting of that final variable.